### ISSUE TITLE
Wrong softmax for attention maps?

### ISSUE THREAD

**Issue by nguyenvo09** [gjylt/DoubleAttentionNet#3]
I think this line is wrong. 
```softmaxB = self.softmax(tmpB).view( batch, self.c_n, self.K*h*w ).permute( 0, 2, 1)  #batch, self.K*h*w, self.c_n```

It should be:
```softmaxB = F.softmax(tmpB, dim = -1).view( batch, self.c_n, self.K*h*w )``` It means we softmax over the `self.K * h * w` (i.e. attention maps)
Any thoughts? 
Solved.
